ocaml-lsp-server/src/hover_req.ml

```diff
@@ -115,7 +167,17 @@ let handle server { HoverParams.textDocument = { uri }; position; _ } mode =
                 ~field:(fun td ->
                   Option.map td.hover ~f:(fun h -> h.contentFormat))
             in
-            format_contents ~syntax ~markdown ~typ ~doc:documentation
+            let syntax_doc_name = syntax_documentation.name in
+            let syntax_doc_des = syntax_documentation.description in
+            let syntax_doc_url = syntax_documentation.documentation in
+            format_contents
+              ~syntax
+              ~markdown
+              ~typ
+              ~doc:documentation
+              ~syntax_doc_name
+              ~syntax_doc_des
+              ~syntax_doc_url
```

Comment:
There is no need to pass each field separately here. You can pass the whole record and destruct it later.

Also, there is sometimes no syntax documentation so you should pass a `Query.protocol.info option` or simply keep the `` `No_documentation / `Found _ `` values returned by merlin (as said in another comment).